src/AlertsManagement/AlertsManagement/ActionRuleCommands/SetAzureActionRule.cs

```diff
@@ -0,0 +1,420 @@
+                ParameterSetName = BySimplifiedFormatSuppressionActionRuleParameterSet,
+                HelpMessage = "Resource Group Name")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Action rule name
+        /// </summary>
+        [Parameter(Mandatory = true,
+                ParameterSetName = ByJsonFormatActionRuleParameterSet,
+                HelpMessage = "Action rule Name")]
+        [Parameter(Mandatory = true,
+                ParameterSetName = BySimplifiedFormatSuppressionActionRuleParameterSet,
+                HelpMessage = "Action rule Name")]
+        [Parameter(Mandatory = true,
+                ParameterSetName = BySimplifiedFormatActionGroupActionRuleParameterSet,
+                HelpMessage = "Action rule Name")]
+        [ValidateNotNullOrEmpty]
+        [Alias("ResourceId")]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Action rule Json
+        /// </summary>
+        [Parameter(Mandatory = true,
+                ParameterSetName = ByJsonFormatActionRuleParameterSet,
+                HelpMessage = "Action rule Json format")]
+        [ValidateNotNullOrEmpty]
+        public string ActionRule { get; set; }
+
+        /// <summary>
+        /// Action rule simplified format : Description
+        /// </summary>
+        [Parameter(Mandatory = false,
+                ParameterSetName = BySimplifiedFormatSuppressionActionRuleParameterSet,
+                HelpMessage = "Description of Action Rule")]
+        [Parameter(Mandatory = false,
+                ParameterSetName = BySimplifiedFormatActionGroupActionRuleParameterSet,
+                HelpMessage = "Description of Action Rule")]
+        public string Description { get; set; }
+
+        /// <summary>
+        /// Action rule simplified format : Status
+        /// </summary>
+        [Parameter(Mandatory = true,
+                ParameterSetName = BySimplifiedFormatActionGroupActionRuleParameterSet,
+                HelpMessage = "Status of Action Rule.")]
+        [Parameter(Mandatory = true,
+                ParameterSetName = BySimplifiedFormatSuppressionActionRuleParameterSet,
+                HelpMessage = "Status of Action Rule.")]
+        [ValidateNotNullOrEmpty]
+        [PSArgumentCompleter("Enabled", "Disabled")]
+        public string Status { get; set; }
+
+        /// <summary>
+        /// Action rule simplified format : Scope Type
+        /// </summary>
+        [Parameter(Mandatory = true,
+                ParameterSetName = BySimplifiedFormatActionGroupActionRuleParameterSet,
+                HelpMessage = "Scope Type")]
+        [Parameter(Mandatory = true,
+                ParameterSetName = BySimplifiedFormatSuppressionActionRuleParameterSet,
+                HelpMessage = "Scope Type")]
+        [ValidateNotNullOrEmpty]
+        [PSArgumentCompleter("Resource", "ResourceGroup")]
+        public string ScopeType { get; set; }
```

Comment:
We do not support mixed list of Resource and ResourceGroup. It could be either Resource or ResourceGroup list. Our api also expects this property explicitly currently. Do we want to take extra overhead for detecting the type ourselves?

Reply:
Makes sense. Committed.